packages/atomic/src/components/ipx/atomic-ipx-button/atomic-ipx-button.tsx

```diff
@@ -0,0 +1,88 @@
+import {Component, Element, h, Prop} from '@stencil/core';
+import CloseIcon from '../../../images/close.svg';
+import SearchIcon from '../../../images/search.svg';
+import {Button} from '../../common/button';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-ipx-button',
+  styleUrl: './atomic-ipx-button.pcss',
+  shadow: true,
+})
+export class AtomicIPXButton {
+  @Element() public host!: HTMLElement;
+
+  /**
+   * The label that will be shown to the user.
+   */
+  @Prop({reflect: true}) public label = 'Workplace Search';
+
+  /**
+   * Whether the IPX layout is open.
+   */
+  private isOpen = false;
+
+  private async onClick() {
+    this.isOpen ? this.close() : this.open();
+    this.isOpen = !this.isOpen;
```

Comment:
ca ferais du sens de setté isOpen dans les méthodes respectives je crois!

Reply:
🤔 it's more about the 1 line vs 2.